framework/aster-frame/src/arch/x86/tdx_guest.rs

```diff
@@ -87,3 +150,279 @@ fn handle_io(trapframe: &mut dyn TdxTrapFrame, ve_info: &tdcall::TdgVeInfo) -> b
     };
     true
 }
+
+fn is_shared_gpa(gpa: u64) -> bool {
+    (gpa & (1 << SHARED_BIT)) != 0
+}
+
+fn handle_mmio(trapframe: &mut dyn TdxTrapFrame, ve_info: &TdgVeInfo) -> Result<usize, MmioError> {
+    // Get instruction
+    let instr = decode_instr(trapframe.rip())?;
+
+    // Decode MMIO instruction
+    let (mmio, size) = decode_mmio(&instr);
+    if mmio.is_none() {
+        return Err(MmioError::DecodeFailed);
+    }
+    let mmio_size = match size {
+        1 => IoSize::Size1,
+        2 => IoSize::Size2,
+        4 => IoSize::Size4,
+        8 => IoSize::Size8,
+        _ => unreachable!(),
```

Comment:
So why is this branch unreachable? If all possible values are 1, 2, 4, 8, why not let the `decode_mmio` method return `(Instruction, IoSize)` so that you don't have to write a unreachable branch and add some comments to explain yourself.